pkg/adapter/db.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adapter
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/containerd/containerd/metadata"
+	bolt "go.etcd.io/bbolt"
+)
+
+var (
+	bucketKeyVersion       = []byte("v1")
+	bucketKeyObjectContent = []byte("content") // stores content references
+	bucketKeyObjectBlob    = []byte("blob")    // stores content links
+
+	bucketKeySize = []byte("size")
+
+	MB = 1024 * 1024
+)
+
+// contentSize return the size of local caches, MB
+func contentSize(db *metadata.DB) (float64, error) {
```

Comment:
Can we wrap the `db` into a struct like `Content`? So that we can provide below methods:

```
func Size() (int64, error)
func GC() error
```